src/version.h

```diff
@@ -11,6 +11,9 @@
 
 static const int PROTOCOL_VERSION = 170002;
```

Comment:
Bug is here: this PR is changing the way the network protocol behaves, but isn't bumping the protocol version itself. The `wallet_overwintertx` is hanging while trying to sync blocks after the first Overwinter block is mined (and the disconnection code activates), because the peers are all advertising an older protocol version.

The protocol version needs to be bumped here instead of in #2970.

Reply:
Hmm, this is not the only bug: I bumped the version, and the nodes now stay connected, but they still don't synchronize.

Reply:
Protocol bug is definitely part of it though, as without changing it, I see this in node 0 debug.log:
```
2018-02-22 16:03:25 UpdateTip: new best=03812205187bf04a9e0372b15ea1ca25b4a75c4bb49be5297da2ded994ff10fe  height=200  log2_work=11.738515  tx=205  date=2018-02-22 16:03:24 progress=1.000000  cache=0.1MiB(202tx)
2018-02-22 16:03:25 AddToWallet 606363805d339bd8ee86509f165edafd47e312a2c29fac30df25c392b54aa0b9  new
2018-02-22 16:03:25 AddToWallet c8c42314b51a406bfecc42913b3c19a0304b8d8c604700dd9201ab2188fc6b1f  update
2018-02-22 16:03:25 AddToWallet 52f0c09d55acc4fdf5644d5d45aa67522cd4a852002bd3fb8b13290f5e54c79f  update
2018-02-22 16:03:25 keypool return 5
2018-02-22 16:03:25 peer=1 using obsolete version 170002; disconnecting
2018-02-22 16:03:25 ProcessMessages(getheaders, 645 bytes) FAILED peer=1
2018-02-22 16:03:25 peer=3 using obsolete version 170002; disconnecting
2018-02-22 16:03:25 ProcessMessages(getheaders, 645 bytes) FAILED peer=3
2018-02-22 16:03:25 peer=5 using obsolete version 170002; disconnecting
2018-02-22 16:03:25 ProcessMessages(getheaders, 645 bytes) FAILED peer=5
```